consensus/replay.go

```diff
@@ -417,7 +419,7 @@ func (h *Handshaker) replayBlocks(state sm.State, proxyApp proxy.AppConns, appBl
 	for i := appBlockHeight + 1; i <= finalBlock; i++ {
 		h.logger.Info("Applying block", "height", i)
 		block := h.store.LoadBlock(i)
-		appHash, err = sm.ExecCommitBlock(proxyApp.Consensus(), block, h.logger, state.LastValidators, h.stateDB)
+		appHash, err = sm.ExecCommitBlock(proxyApp.Consensus(), block, h.logger, h.stateDB)
```

Comment:
Huh. I was pretty convinced we weren't going to be able to fix this issue without eliminating `ExecCommitBlock` and using `ApplyBlock` properly.  ExecCommitBlock doesn't mutate the `State` - so how do the updates propagate?

Reply:
In replay, we should not update the state, so I think retain ExecCommitBlock is reasonable.  Replay could call or not call ApplyBlock, if call, then state will be saved, then propagate the updates, if not, no updates made to state, so no need to propagate.

Reply:
I see. We already know what the validators are supposed to be in the Tendermint state, we just need to fetch that for the BeginBlock info.